orquesta/tests/unit/conduct_mock/test_conduct_mock.py

```diff
@@ -0,0 +1,221 @@
+            expected_workflow_status=expected_workflow_status,
+            expected_output=expected_output,
+        )
+        # will throw
+        self.assertRaises(exc.StatusEquality, mock.assert_conducting_sequences)
+
+    def test_sequential_route_equality_throws(self):
+        wf_name = "sequential"
+
+        expected_task_seq = ["task1", "task2", "task3", "continue"]
+        expected_routes = [[], ["wrong"]]
+
+        mock_results = [
+            "Stanley",
+            "All your base are belong to us!",
+            "Stanley, All your base are belong to us!",
+        ]
+
+        expected_output = {"greeting": mock_results[2]}
+
+        self.assert_spec_inspection(wf_name)
+        wf_def = self.get_wf_def(wf_name)
+        wf_spec = self.spec_module.instantiate(wf_def)
+        mock = conduct_mock.WorkflowConductorMock(
+            wf_spec,
+            expected_task_seq,
+            inputs={"name": "Stanley"},
+            expected_routes=expected_routes,
+            mock_results=mock_results,
+            expected_output=expected_output,
+        )
+        # will throw
+        self.assertRaises(exc.RouteEquality, mock.assert_conducting_sequences)
+
+    def test_sequential_task_equality_throws(self):
+        wf_name = "sequential"
+
+        expected_task_seq = ["task1", "task2", "task3", "continue", "exception"]
+
+        mock_results = [
+            "Stanley",
+            "All your base are belong to us!",
+            "Stanley, All your base are belong to us!",
+        ]
+
+        expected_output = {"greeting": mock_results[2]}
+
+        self.assert_spec_inspection(wf_name)
+        wf_def = self.get_wf_def(wf_name)
+        wf_spec = self.spec_module.instantiate(wf_def)
+        mock = conduct_mock.WorkflowConductorMock(
+            wf_spec,
+            expected_task_seq,
+            inputs={"name": "Stanley"},
+            mock_results=mock_results,
+            expected_output=expected_output,
+        )
+        # will throw
+        self.assertRaises(exc.TaskEquality, mock.assert_conducting_sequences)
+
+    def test_sequential_task_output_throws(self):
+        wf_name = "sequential"
+
+        expected_task_seq = ["task1", "task2", "task3", "continue"]
+
+        mock_results = [
+            "Stanley",
+            "All your base are belong to us!",
+            "Stanley, All your base are belong to us!",
+        ]
+
+        expected_output = {"greeting": mock_results[2], "exception": True}
+
+        self.assert_spec_inspection(wf_name)
+        wf_def = self.get_wf_def(wf_name)
+        wf_spec = self.spec_module.instantiate(wf_def)
+        mock = conduct_mock.WorkflowConductorMock(
+            wf_spec,
+            expected_task_seq,
+            inputs={"name": "Stanley"},
+            mock_results=mock_results,
+            expected_output=expected_output,
+        )
+        # will throw
+        self.assertRaises(exc.OutputEquality, mock.assert_conducting_sequences)
```

Comment:
I don't see any support for more advanced use cases such as branching, join, with items, etc.  If we also get the unit tests to use this mock conductor for testing, then those test cases will be covered.

Reply:
This will get fixed as part of class WorkflowConductorMock replacing WorfkflowConductorTest as above.